src/apps/microsoft/hyperv/2012/local/mode/nodeintegrationservice.pm

```diff
@@ -235,16 +235,34 @@ sub manage_selection {
         my $services = (ref($node->{services}) eq 'ARRAY') ? $node->{services} : [ $node->{services} ];
 
         foreach my $service (@$services) {
+            my $primary_status;
+            my $secondary_status;
+            if ( defined($service->{primary_operational_status}) ) {
```

Comment:
This if could have been a sub with one parameter and you could use it 2 time

Reply:
You're right. New commit incoming

Reply:
Done.